crates/turbopack-ecmascript/src/lib.rs

```diff
@@ -158,15 +167,40 @@ impl EcmascriptModuleAssetVc {
     }
 
     #[turbo_tasks::function]
-    pub async fn analyze(self) -> Result<AnalyzeEcmascriptModuleResultVc> {
+    pub async fn analyze(self, failsafe: bool) -> Result<AnalyzeEcmascriptModuleResultVc> {
         let this = self.await?;
-        Ok(analyze_ecmascript_module(
+        let result = analyze_ecmascript_module(
             this.source,
             self.as_resolve_origin(),
             Value::new(this.ty),
             this.transforms,
             this.compile_time_info,
-        ))
+        );
+        if failsafe {
+            let result_value = result.await?;
+            if result_value.successful {
+                this.last_successful_analyse.set(Some((
+                    result,
+                    result_value.references.await?,
+                    result_value.exports.await?,
+                )));
+            } else {
+                if let Some((last_operation, last_references, last_exports)) =
+                    &*this.last_successful_analyse.get()
+                {
+                    let last_operation: turbo_tasks::RawVc = last_operation.into();
+                    last_operation.connect();
```

Comment:
Create and link a task to either: 1. get rid of this, or 2. create a better turbo tasks primitive to do this.

Reply:
https://linear.app/vercel/issue/WEB-699/rawvcconnect-should-be-exposed-in-a-different-way